components/builder-api/src/server/resources/origins.rs

```diff
@@ -213,7 +213,22 @@ fn create_origin(req: HttpRequest,
                                  default_package_visibility: &dpv, };
 
     match Origin::create(&new_origin, &*conn).map_err(Error::DieselError) {
-        Ok(origin) => HttpResponse::Created().json(origin),
+        Ok(origin) => {
+            match OriginAudit::audit(&OriginAudit { operation:      OriginOperation::OriginCreate,
+                                                    origin:         &body.0.name,
+                                                    target_object:  &body.0.name,
+                                                    requester_id:   session.get_id() as i64,
+                                                    requester_name: session.get_name(), },
+                                     &*conn)
+            {
+                Ok(_) => HttpResponse::Created().json(origin),
```

Comment:
Just minor note that you can simplify/DRY this by just doing an if instead of a match.
Eg something like:
```
if let Err(err) = OriginAudit::audit(...) { debug!("....") }
HttpResponse::Created().json(origin)
```
Similarly in the other sections below

Reply:
Brilliant! I wasn't 100% happy with that but couldn't find a better way. `if let` to the rescue!